File: src/main/java/com/ftn/cdss/model/MedicalChart.java
package com.ftn.cdss.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Entity
@Table
public class MedicalChart {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Integer ssn;

    private String name;

    private String surname;
}

File: frontend/src/app/domain/MedicalChart.ts
import {Allergy} from "./Allergy";

export class MedicalChart {

  public id: number;
  public ssn: number;
  public name: string;
  public surname: string;
  public diagnosis: number[];
  public allergies: Allergy[];

  constructor();

  constructor(id: number,
              ssn: number,
              name: string,
              surname: string,
              diagnosis: number[],
              allergies:Allergy[]);

  constructor(id?: number,
              ssn?: number,
              name?: string,
              surname?: string,
              diagnosis?: number[],
              allergies?:Allergy[]) {
    this.id = id;
    this.ssn = ssn;
    this.name = name;
    this.surname = surname;
    this.diagnosis = diagnosis;
    this.allergies = allergies;
  }
}
